### ISSUE TITLE
Edit button on cells should be guessed by the field's type

### ISSUE THREAD

**Issue by lucasbordeau** [twentyhq/twenty#1834]
## Scope & Context

We have an edit button on cells that allows to enter edit mode.

<img width="145" alt="Capture d’écran 2023-10-03 à 11 44 55" src="https://github.com/twentyhq/twenty/assets/26528466/27dee37a-68ca-426e-bf06-bd51873101c7">

## Current behavior

Currently, it is shown if the column definition or field definition specifies it : 

```typescript
export type FieldDefinition<T extends FieldMetadata> = {
  key: string;
  name: string;
  Icon?: IconComponent;
  type: FieldType;
  metadata: T;
  useEditButton?: boolean; // <=== Here
};
```

## Expected behavior

Edit buttons are more related to the type of the field, for example an URL in a cell is clickable and can thus prevent us from entering edit mode.

We should have a way to tie each field metadata type to a boolean which tells if we should show an edit button or not.

## Technical inputs

This could be kept as an hardcoded constant, because it won't change after we've defined it for each type.

**Comment by AyushAgrawal-A2** [comment]
May I take this one?

**Comment by AyushAgrawal-A2** [comment]
@lucasbordeau 
In PR #1746 I have already switched to setting buttonIcon on FiledDefinition and now whatever Icon is set on FiledDefinition shows up on hover.
![image](https://github.com/twentyhq/twenty/assets/54364088/ed79471b-3b54-42dc-be46-e8fc8c810f7c)

Additionally, a boolean value useEditButton is set for FieldRelationMetadata, which is not being used anywhere.
![image](https://github.com/twentyhq/twenty/assets/54364088/4b50ce35-cdeb-43e1-94f7-7dbf8f2e9931)

So to summarize, useEditButton is not being used anywhere.
Any Icon set on FieldDefinition will show up on hover. Click behavior of this Icon is governed by openTableCell.

Now, in order to define Icon on a per FieldMetadata basis shall I move the Icon component to FieldMetadata from FieldDefinition ?

**Comment by lucasbordeau** [comment]
We want to go one step further and hard code when a button should show up for each type.

For example every URL type should have an edit button because the only action we want to allow is to edit it, and the icon should be a pen each time.

We don't need to pass it to FieldDefinition or FieldURLMetadata because we now it's required for every URL cell.

You can already make a first implementation with URL field type and we'll ask @Bonapara for every type that should have a pen edit button.
